web/src/Api/Endpoint.elm

```diff
@@ -48,10 +50,46 @@ type StudentEndpoint
     = StudentEndpoint
 
 
+type StudentLogoutEndpoint
+    = StudentLogoutEndpoint
+
+
+type StudentResearchConsentEndpoint
+    = StudentResearchConsentEndpoint
+
+
+type StudentUsernameValidEndpoint
+    = StudentUsernameValidEndpoint
+
+
+studentValidUsernameEndpoint : Api.Config.Config -> Endpoint StudentUsernameValidEndpoint
+studentValidUsernameEndpoint config =
+    Endpoint
+      StudentUsernameValidEndpoint
+        (Api.Config.restApiUrl config)
+        ["api", "username"]
+        Nothing
+
 studentEndpoint : Profile.ProfileID -> Api.Config.Config -> Endpoint StudentEndpoint
 studentEndpoint profileId config =
     Endpoint
       StudentEndpoint
         (Api.Config.restApiUrl config)
         ["api", "student", String.fromInt (Profile.profileID profileId)]
+        Nothing
+
+studentResearchConsentEndpoint : Profile.ProfileID -> Api.Config.Config -> Endpoint StudentResearchConsentEndpoint
+studentResearchConsentEndpoint profileId config =
+    Endpoint
+      StudentResearchConsentEndpoint
+        (Api.Config.restApiUrl config)
+        ["api", "student", String.fromInt (Profile.profileID profileId), "consent_to_research"]
+        Nothing
+
+studentLogoutEndpoint : Api.Config.Config -> Endpoint StudentLogoutEndpoint
+studentLogoutEndpoint config =
+    Endpoint
+      StudentLogoutEndpoint
+        (Api.Config.restApiUrl config)
+        ["api", "student", "logout"]
         Nothing
```

Comment:
We might want to use `Api.logout` when logging out users instead of adding an endpoint for it. 

Part of the logout process involves removing the token form `localStorage` which means calling out over a `port` to JavaScript. If we make an API call to the server to logout, it should be after we have removed the token and probably in JavaScript so that it happens right away. This makes it so that the token is the single source of truth about whether a user is logged in or not.

We might also not need to make a call to the server to log users out! When their JWT expires, they should be implicitly logged out as far the frontend and backend are concerned.

Reply:
If we want the option to let the server know we've logged out I'd presume we'd want to keep these endpoints.  We can take out calls to them for now, however and just remove the token.  I'm fine with that.

Reply:
Sounds good. Let's go with no call for now.